### ISSUE TITLE
Folder structure readability 

### ISSUE THREAD

**Issue by Salvatore-tech** [pizza11x/android-MVVM-architecture#23]
I'm proposing a new folder structure that is possibly more readable and less deep.
The aim is to enhance the separation of source code files and non-source files and shorten the path to the files. The following folder are intended to start from the project root.

1. manifests: AndroidManifest.xml
2. java: java and kotlin source code, test files should also be placed under it and marked as "test folder" in IDE
3. res: all non-code files, you already created the subfolders "drawable", "layout", "mipmap" and "values"
4. Gradle Scripts: files for build configuration ("build.gradle", "settings.gradle"...)

**Comment by pizza11x** [comment]
When you create a new Android project, the standard structure that the idea generates is the following:

- Manifest: with the only manifest file

- Java: with three subdirectories, two for testing and one for source code:
That one non-test folder is the actual root folder.
it is also standard to have in the root folder:
"com or it" dot "company / user name" dot "project name"

- Res: Android studio automatically generates folders, even if empty. It is good practice not to delete them until you are certain that you have no items inside.

- Gradle: there are several gradle files within a project:
those of the whole project where you insert the dependencies of the whole project, the app and the individual modules. and it is right that they are paid in the various levels.

My personal choices were to create three folders inside my root folder (androidMVVMarchitecture), one for each layer of the MVVM architecture:
**Data**, **Domain** and **App** or **Presentation** (I decided to call it presentation so as not to confuse it with the "APP" folder generated by Android studio.)
Within the three folders we find subfolders that define the components of each level of the architecture.
_Es -> in domain we will find: models, usecases, and repo_
In my case, since I didn't create any files for the repo folder, I didn't create the same folder accordingly.

In addition, I created a **utils** folder at the same level as the three layer folders. But that will most likely become an External Module which will then be invoked through dependencies.

Surely the organization of the folders is to be improved, and will be done towards the completion of the library, so as to have a more precise picture of the various classes.

Thank you for the suggestion, I will leave the issue open until it is time to improve the organization
